### PR TITLE
Fixed `no PropertyInfo extractor` problem of docs

### PR DESCRIPTION
**Pull request by gustawdaniel**
Problem is described here

> https://github.com/api-platform/core/issues/2563

It occurs when you not define properties in Entity.

Additionally added `@ApiProperty(identifier=true)` because of error "No identifiers defined for resource of type \"App\\Entity\\Book\""

Also removed services registration because of symfony has now autowiring by default.

Added getter and setter in DTO classes because of it allow to generate models in docs

### CODE REVIEW

File: core/dto.md [hunk 1]
```diff
@@ -23,6 +24,12 @@ use App\Dto\BookOutput;
  */
 final class Book
 {
+    /**
+     * At least one property should be identifier
+     * @ApiProperty(identifier=true)
```
Comment (teohhanhui) [inline core/dto.md]: It makes much more sense to use `$isbn` as identifier.

Comment (gustawdaniel) [reply core/dto.md]: I agree.

File: core/dto.md [hunk 2]
```diff
@@ -46,7 +53,12 @@ We have the following `BookInput`:
 namespace App\Dto;
 
 final class BookInput {
-  public $isbn;
+    public $isbn;
+
+    /*
+     * Setter is required to generate documentation of example model
+     */
+    public function setIsbn(string $isbn) { $this->isbn = $isbn; }
```
Comment (dunglas) [inline core/dto.md]: It shouldn't be the case. Just adding the following PHPDoc on the property should be enough:
```
/**
  * @var string The ISBN number
  */
```

Can you test this please?

File: core/dto.md [hunk 3]
```diff
@@ -60,36 +72,46 @@ namespace App\DataTransformer;
 
 use ApiPlatform\Core\DataTransformer\DataTransformerInterface;
 use App\Dto\BookInput;
+use App\Entity\Book;
 
 final class BookInputDataTransformer implements DataTransformerInterface
 {
     /**
```
Comment (dunglas) [inline core/dto.md]: You should just totally remove this docblock in the example. It brings nothing.

File: core/dto.md [hunk 4]
```diff
@@ -60,36 +72,46 @@ namespace App\DataTransformer;
...
+     * @param string $to
+     * @param array $context
+     * @return Book
      */
     public function transform($data, string $to, array $context = [])
     {
-        $book = new Book();
-        $book->isbn = $data->isbn;
-        return $book;
+        if($data instanceof BookInput) {
```
Comment (dunglas) [inline core/dto.md]: What if it's not a BookInput? Also, you should use a guard clause, and fix CS.

File: core/dto.md [hunk 5]
```diff
@@ -60,36 +72,46 @@ namespace App\DataTransformer;
...
+            $book = new Book();
+            $book->isbn = $data->isbn;
+            $book->name = strrev($data->isbn); // there can be called service that calculate lacking fields, `strrev` is used for example
+            return $book;
+        }
     }
 
     /**
-     * {@inheritdoc}
+     * BookInput -> Book
```
Comment (dunglas) [inline core/dto.md]: This dockblock brings nothing, and can be confusing, please remove it

File: core/dto.md [hunk 6]
```diff
@@ -128,26 +155,39 @@ use App\Entity\Book;
 final class BookOutputDataTransformer implements DataTransformerInterface
 {
     /**
-     * {@inheritdoc}
+     * Convert Book into BookOutput
```
Comment (dunglas) [inline core/dto.md]: To remove

File: core/dto.md (hunk 2 above)
Comment (gustawdaniel) [reply core/dto.md]: Yes. I tested. It works exactly like yo wrote.

File: core/dto.md (hunk 4 above)
Comment (gustawdaniel) [reply core/dto.md]: What is CS? I can simply remove this `if`

Comment (teohhanhui) [reply core/dto.md]: CS = coding standards / coding style. There are multiple CS issues in your changes. Here, for example, it must be:

```php
if (...) {
```

not:

```php
if(...) {
```

**Comment by smxsm** [comment]
Why isn't this merged? The official docs are still wrong and the code there is still throwing this error ...

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (alanpoulain) [inline .gitignore]: Can you remove this file?

**Comment by alanpoulain** [comment]
It should be rebased and target 2.5 now.

File: core/dto.md [hunk 8]
```diff
@@ -23,6 +24,12 @@ use App\Dto\BookOutput;
  */
 final class Book
 {
+    public $name;
```
Comment (alanpoulain) [inline core/dto.md]: New line?

File: core/dto.md [hunk 9]
```diff
@@ -46,7 +53,10 @@ We have the following `BookInput`:
 namespace App\Dto;
 
 final class BookInput {
```
Comment (alanpoulain) [inline core/dto.md]: New line?

File: core/dto.md (hunk 3 above)
Comment (alanpoulain) [reply core/dto.md]: Has not been done.

**Comment by gustawdaniel** [comment]
I only wanted improve documentation. 

Key point of this pull request was prevent from error `no PropertyInfo extractor`, no spaces after `if`, new lines and `.idea` in `.gitignore`.

I give up and closing this pull request.

**Pull request closed without merging**